corehq/apps/users/dbaccessors/all_commcare_users.py

```diff
@@ -1,18 +1,24 @@
+from itertools import imap
 from dimagi.utils.couch.database import iter_docs
 
 
 def get_all_commcare_users_by_domain(domain):
     """Returns all CommCareUsers by domain regardless of their active status"""
     from corehq.apps.users.models import CommCareUser
-    key = [domain, CommCareUser.__name__]
-    ids = [user['id'] for user in CommCareUser.get_db().view(
-        'domain/docs',
-        startkey=key,
-        endkey=key + [{}],
-        reduce=False,
-        include_docs=False)]
 
-    return [CommCareUser.wrap(user) for user in iter_docs(CommCareUser.get_db(), ids)]
+    def get_ids():
+        for flag in ['active', 'inactive']:
+            key = [flag, domain, CommCareUser.__name__]
+            for user in CommCareUser.get_db().view(
+                'users/by_domain',
+                startkey=key,
+                endkey=key + [{}],
+                reduce=False,
+                include_docs=False
+            ):
+                yield user['id']
```

Comment:
I couldn't figure out a way to get all users in a domain in one query (which is why I imagine this used `domain/docs` in the first place), but two isn't bad.  It's either this or filter out `CouchUser-Deleted` docs in memory.